In [11]:
import numpy as np
import pandas as pd

In [12]:
df = pd.read_csv("student-mat.csv")
df.head(3)

school sex  age address famsize Pstatus  Medu  Fedu     Mjob     Fjob  ...  \
0     GP   F   18       U     GT3       A     4     4  at_home  teacher  ...   
1     GP   F   17       U     GT3       T     1     1  at_home    other  ...   
2     GP   F   15       U     LE3       T     1     1  at_home    other  ...   

  famrel freetime  goout  Dalc  Walc health absences G1 G2  G3  
0      4        3      4     1     1      3        6  5  6   6  
1      5        3      3     1     1      3        4  5  5   6  
2      4        3      2     2     3      3       10  7  8  10  

[3 rows x 33 columns]

In [13]:
len(df)

395

#### Add a boolean column called grade_A noting if a student achieved 80% or higher as a final score.

In [14]:
df['grade_A'] = np.where(df['G3']*5 >= 80, 1, 0)

#### Make another boolean column called high_abscences with a value of 1 if a student misses 10 or more classes.

In [15]:
df['high_absences'] = np.where(df['absences'] >= 10, 1, 0)

#### Add one more column to make building the pivot table easier

In [16]:
df['count'] = 1

In [17]:
df

school sex  age address famsize Pstatus  Medu  Fedu      Mjob      Fjob  \
0       GP   F   18       U     GT3       A     4     4   at_home   teacher   
1       GP   F   17       U     GT3       T     1     1   at_home     other   
2       GP   F   15       U     LE3       T     1     1   at_home     other   
3       GP   F   15       U     GT3       T     4     2    health  services   
4       GP   F   16       U     GT3       T     3     3     other     other   
..     ...  ..  ...     ...     ...     ...   ...   ...       ...       ...   
390     MS   M   20       U     LE3       A     2     2  services  services   
391     MS   M   17       U     LE3       T     3     1  services  services   
392     MS   M   21       R     GT3       T     1     1     other     other   
393     MS   M   18       R     LE3       T     3     2  services     other   
394     MS   M   19       U     LE3       T     1     1     other   at_home   

     ... Dalc Walc  health  absences  G1  G2  G3 grade_A high_absences count  
0    ...    1    1       3         6   5   6   6       0             0     1  
1    ...    1    1       3         4   5   5   6       0             0     1  
2    ...    2    3       3        10   7   8  10       0             1     1  
3    ...    1    1       5         2  15  14  15       0             0     1  
4    ...    1    2       5         4   6  10  10       0             0     1  
..   ...  ...  ...     ...       ...  ..  ..  ..     ...           ...   ...  
390  ...    4    5       4        11   9   9   9       0             1     1  
391  ...    3    4       2         3  14  16  16       1             0     1  
392  ...    3    3       3         3  10   8   7       0             0     1  
393  ...    3    4       5         0  11  12  10       0             0     1  
394  ...    3    3       5         5   8   9   9       0             0     1  

[395 rows x 36 columns]

In [18]:
df = df[['grade_A', 'high_absences', 'count']]
df.head()

grade_A  high_absences  count
0        0              0      1
1        0              0      1
2        0              1      1
3        0              0      1
4        0              0      1

In [20]:
pd.pivot_table(df,
              values = 'count',
              index = ['grade_A'],
              columns = ['high_absences'],
              aggfunc = np.size,
              fill_value = 0
              )

high_absences    0   1
grade_A               
0              277  78
1               35   5

#### In our case, P(A) is the probability of a grade of 80% or greater. P(B) is the probability of missing 10 or more classes. P(A|B) is the probability of a 80%+ grade, given missing 10 or more classes.

Calculations of parts:

###### P(A)
= (35 + 5)/(35 + 5 + 277 + 78) <br>
= 0.10126582278481013

###### P(B)
= (78 + 5)/(35 + 5 + 277 + 78) <br>
= 0.21012658227848102
